src/data/data-provider.js

```diff
@@ -12,20 +16,278 @@ export function getSchools (country, camp) {
+              } else {
+                values[school] = getZeroLessonData()
+              }
+            }
+          }
+        }
+      }
+    }
   }
 
-  return {
-    months: Object.keys(values),
+  values = {
+    labels: Object.keys(values),
     lessons: Object.values(values)
   }
+
+  return values
+}
+
+export function getLessonsByTopics (country, camp, school, year) {
+  let values = data.children[country].children[camp].children[school].children
+  let allMonthlyDataByYears = Object.values(values)
+
+  const allTopics = Object.keys(values)
+  let existingTopicsInYear = {}
+
+  allTopics.forEach((el, index) => {
+    const monthlyData = allMonthlyDataByYears[index].values[year]
+    existingTopicsInYear[el] = monthlyData || getZeroLessonData()
+  })
+
+  return {
+    labels: Object.keys(existingTopicsInYear),
+    lessons: Object.values(existingTopicsInYear)
+  }
+}
+
+// set year select box options
+export function setYearSelectBox (country, camp, school) {
+  const totalDataByCountry = Object.values(data.children)
+  let yearlyIndex = []
+  let uniqueYear = []
+
+  if (country === undefined) {
+    totalDataByCountry.forEach(el => {
+      const totalData = el.values
+      yearlyIndex.push(...Object.keys(totalData))
+    })
+    uniqueYear = yearlyIndex.filter((item, index) => yearlyIndex.indexOf(item) === index).sort((a, b) => b - a)
+  }
+
+  return uniqueYear.slice(1, uniqueYear.length)
 }
+
+// for ICT
+export function getIctCountries () {
+  const countries = ICT_STUDENT_DATA
+  return Object.keys(countries.children)
+}
+
+export function getAvgPercentage (absolute) {
+  const MAX_ICT = 21
+  const percentage = absolute / MAX_ICT * 100
+  return percentage
+}
+
+export function getIctSchoolList () {
+  const list = Object.keys(ICT_STUDENT_DATA.children.Tanzania.children.Nyarugusu.children)
+  return list
+}
+
+export function getStudentIctSchoolAvg (school, type, year) {
+  let percentage = ''
+  if (year === 'Base') {
+    const base = ICT_STUDENT_DATA.children.Tanzania.children.Nyarugusu.children[school].values[type].base_avg
+    percentage = getAvgPercentage(base).toFixed(0)
+  } else if (year === 'End') {
+    const end = ICT_STUDENT_DATA.children.Tanzania.children.Nyarugusu.children[school].values[type].end_avg
+    percentage = getAvgPercentage(end).toFixed(0)
+  }
+  return percentage
+}
+
+export function getStudentAvgAcrossSchools (type, year) {
+  let percentage = ''
+  if (year === 'Base') {
+    const base = ICT_STUDENT_DATA.values[type].base_avg
+    percentage = getAvgPercentage(base).toFixed(0)
+  } else if (year === 'End') {
+    const end = ICT_STUDENT_DATA.values[type].end_avg
+    percentage = getAvgPercentage(end).toFixed(0)
+  }
+  return percentage
+}
+
+export function getTeacherIctSchoolAvg (school, type, year) {
+  let percentage = ''
+  if (year === 'Base') {
+    const base = ICT_TEACHER_DATA.children.Tanzania.children.Nyarugusu.children[school].values[type].base_avg
```

Comment:
Can we extract the object as a constant default value, because we're using it across different functions.

Reply:
Can you more clarify about this comment?
I'm not using this variable other than in this function.

Reply:
I've noticed that this phrase:
children.Tanzania.children.Nyarugusu.children
has 39 occurences in the code, so I thought about using it as a constant (default path) instead of copy paste. 
It's not super important, just a code style improvement.

Reply:
Aha, I understood! I'll fix that as you said